demo-circuit/src/naive_threshold_sig/mod.rs

```diff
@@ -267,56 +291,57 @@ pub fn generate_parameters(max_pks: usize) -> Result<Parameters<MNT4>, Synthesis
         _field:                   PhantomData
     };
 
-    let params = generate_random_parameters::<MNT4, _, _>(c, &mut rng);
-    params
+    let universal_srs = MarlinInst::universal_setup(2usize.pow(20), 2usize.pow(20), 2usize.pow(20), &mut rng).unwrap();
+
+    Ok(MarlinInst::index(
+        &universal_srs,
+        c,
+    ).unwrap())
 }
 
 #[cfg(test)]
 mod test {
     use super::*;
-    use algebra::{curves::mnt4753::MNT4, BigInteger768, ProjectiveCurve};
+    use algebra::{BigInteger256, ProjectiveCurve};
     use primitives::{
         crh::FieldBasedHash,
         signature::{
             FieldBasedSignatureScheme, schnorr::field_based_schnorr::FieldBasedSchnorrSignatureScheme,
         },
     };
-    use proof_systems::groth16::{
-        Parameters,
-        Proof, create_random_proof,
-        prepare_verifying_key, verify_proof,
-    };
+    use poly_commit::Error as PCError;
+    use marlin::{Proof, Error as MarlinError};
     use rand::{
-        Rng, rngs::OsRng
+        Rng, rngs::OsRng, thread_rng
     };
 
-    type SchnorrSig = FieldBasedSchnorrSignatureScheme<MNT4Fr, MNT6G1Projective, MNT4PoseidonHash>;
+    type SchnorrSig = FieldBasedSchnorrSignatureScheme<Fr, Projective, PoseidonHash>;
 
     fn generate_test_proof(
         max_pks:                  usize,
         valid_sigs:               usize,
         threshold:                usize,
         wrong_pks_threshold_hash: bool,
         wrong_wcert_sysdata_hash: bool,
-        params:                   Parameters<MNT4>,
-    ) -> Result<(Proof<MNT4>, Vec<MNT4Fr>), SynthesisError> {
+        params:                   (IndexProverKey<Fr, IPAPC>, IndexVerifierKey<Fr, IPAPC>),
```

Comment:
You should need only the prover key for this function no ?

Reply:
Done